### PR TITLE
[CORL-1090] Sections Alpha

### PR DESCRIPTION
**Pull request by wyattjoh**
## What does this PR do?

Adds the alpha for section filtering in the moderation screen. This requires the `SECTIONS` feature flag to be enabled.

## What changes to the GraphQL/Database Schema does this PR introduce?

- Added new `SECTIONS` enum value to the `FEATURE_FLAG` enum
- Added new `SectionFilter` input type
- Added new `section` filter on `Query.comments` edge
- Added new `section` filter on `Query.moderationQueues` edge
- Added new `sections` edge to `Query`
- Added `section` and `siteID` to `ApproveCommentPayload.moderationQueues` edge
- Added `section` and `siteID` to `RejectCommentPayload.moderationQueues` edge
- Added `section` and `siteID` to `Subscription.commentEnteredModerationQueue` edge
- Added `section` and `siteID` to `Subscription.commentLeftModerationQueue` edge

## How do I test this PR?

This feature only currently functions with a feature flag and when in single site mode (only one site is configured):

1. Enable `SECTIONS` feature flag
2. Create a story with a section
3. Post a comment on that story with a section
4. Navigate to moderation
5. Select the section dropdown to select the new section to find the new comment

### CODE REVIEW

**Comment by wyattjoh** [comment]
Note that this feature requires the manual addition of an index:

```js
db.stories.createIndex({ tenantID: 1, "metadata.section": 1 });
```

Once this feature exists the feature flag gate, the index will be added to the codebase.

File: src/core/client/framework/lib/network/createManagedSubscriptionClient.ts [hunk 1]
```diff
@@ -166,6 +184,19 @@ export default function createManagedSubscriptionClient(
...
+        if (process.env.NODE_ENV !== "production") {
+          window.console.debug(
+            `-1 [${requests.length + 1} - 1 = ${
+              requests.length
+            }] dispose called for subscription:`,
+            request.operation?.name,
+            "with variables:",
+            JSON.stringify(request.variables)
+          );
+        }
```
Comment (alinhle) [inline src/core/client/framework/lib/network/createManagedSubscriptionClient.ts:199]: Nice

File: src/core/client/framework/lib/relay/fetch.tsx [hunk 2]
```diff
@@ -49,14 +48,8 @@ export async function fetchQuery<T extends { response: any }>(
...
-): Promise<T["response"][keyof T["response"]]> {
-  const result = await relayFetchQuery(
-    environment,
-    taggedNode,
-    variables,
-    cacheConfig
-  );
-  return extractPayload(result);
+): Promise<T["response"]> {
+  return relayFetchQuery(environment, taggedNode, variables, cacheConfig);
```
Comment (alinhle) [inline src/core/client/framework/lib/relay/fetch.tsx:52]: You didn't like my magic here 😄

File: src/core/server/graph/resolvers/Subscription/commentLeftModerationQueue.ts [hunk 3]
```diff
@@ -37,6 +42,22 @@ export const commentLeftModerationQueue: SubscriptionToCommentLeftModerationQueu
...
+      // specific site.
+      if (siteID && source.siteID !== siteID) {
+        return false;
+      }
+
+      // If we're filtering by section, then only send back comments from the
+      // specific section.
+      if (
+        section &&
+        section.name !== source.section &&
```
Comment (alinhle) [inline src/core/server/graph/resolvers/Subscription/commentLeftModerationQueue.ts]: What about when `section.name === null`? I think `source.section` is `undefined` but never `null`

File: src/core/server/graph/resolvers/Subscription/commentEnteredModerationQueue.ts [hunk 4]
```diff
@@ -38,6 +43,22 @@ export const commentEnteredModerationQueue: SubscriptionToCommentEnteredModerati
...
+      // specific site.
+      if (siteID && source.siteID !== siteID) {
+        return false;
+      }
+
+      // If we're filtering by section, then only send back comments from the
+      // specific section.
+      if (
+        section &&
+        section.name !== source.section &&
```
Comment (alinhle) [inline src/core/server/graph/resolvers/Subscription/commentEnteredModerationQueue.ts]: See above

**Review by alinhle — COMMENTED** [review]
Looks great! Here are some comments. Also it seems the `approved` and `rejected` queue does not follow the  `section` parameter that is passed.

**Comment by wyattjoh** [comment]
Issues should be fixed @cvle!

**Review by alinhle — APPROVED** [review]

**Pull request merged**